### PR TITLE
ddl: fix err check

### PR DESCRIPTION
**Pull request by Tjianke**
### What problem does this PR solve?

Issue Number: #22979  

Problem Summary:

### What is changed and how it works?

What's Changed: added err check

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note.

### CODE REVIEW

**Comment by bb7133** [comment]
Hi @Tjianke , thanks for your contribution!

BTW, since this PR doesn't change any behavior of TiDB('nothing to release), I've changed the release note to 'No release note'.

**Comment by bb7133** [comment]
PTAL @zimulala @xhebox

File: ddl/db_change_test.go [hunk 1]
```diff
@@ -502,14 +511,17 @@ func (s *testStateChangeSuite) TestAppendEnum(c *C) {
...
+	err = checkResult(result, testkit.Rows(expected...))
+	c.Assert(err, IsNil)
 
 	_, err = s.se.Execute(context.Background(), "update t set c2='N' where c4 = 10")
 	c.Assert(err, IsNil)
 	result, err = s.execQuery(tk, "select c2 from t where c4 = 10")
 	c.Assert(err, IsNil)
-	expected = []string{"8 N", "10 N", "11 A"}
-	checkResult(result, testkit.Rows(expected...))
+	// fixed
```
Comment (tangenta) [inline ddl/db_change_test.go:521]:
```suggestion
```

File: ddl/db_change_test.go [hunk 2]
```diff
@@ -1161,7 +1203,11 @@ func (s *testStateChangeSuite) TestParallelDropPrimaryKey(c *C) {
 func (s *testStateChangeSuite) TestParallelCreateAndRename(c *C) {
 	sql1 := "create table t_exists(c int);"
 	sql2 := "alter table t rename to t_exists;"
-	defer s.se.Execute(context.Background(), "drop table t_exists")
+	defer func() {
+		// fixed
```
Comment (tangenta) [inline ddl/db_change_test.go:1207]:
```suggestion
```

File: ddl/db_change_test.go [hunk 3]
```diff
@@ -1253,15 +1301,20 @@ func (s *testStateChangeSuiteBase) testControlParallelExecSQL(c *C, sql1, sql2 s
...
 		c.Assert(err, IsNil)
 	}
-	defer s.se.Execute(context.Background(), "drop table t")
+	defer func() {
+		_, err := s.se.Execute(context.Background(), "drop table t")
+		c.Assert(err, IsNil)
+	}()
 
-	_, err = s.se.Execute(context.Background(), "drop database if exists t_part")
+	// fixed
```
Comment (tangenta) [inline ddl/db_change_test.go:1309]:
```suggestion
```

**Comment by zimulala** [comment]
@Tjianke Please fix the conflict.

**Pull request merged**